internal/oc/extract.go

```diff
@@ -0,0 +1,50 @@
+package oc
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/openshift/assisted-service/pkg/executer"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=extract.go -package=oc -destination=mock_extract.go
+type Extracter interface {
+	Extract(log logrus.FieldLogger, imageIndexPath string, openshiftVersion string, filePath string, pullSecret string) (string, error)
+	ExtractDatabaseIndex(log logrus.FieldLogger, openshiftVersion string, pullSecret string) (string, error)
+}
+
+type extract struct {
+	executer executer.Executer
+	config   Config
+}
+
+func NewExtracter(executer executer.Executer, config Config) Extracter {
+	return &extract{executer, config}
+}
+
+const (
+	imageIndex           = "registry.redhat.io/redhat/redhat-operator-index"
```

Comment:
How would it work for disconnected?
We should derive the index image from the OpenShift image.

Reply:
Why from OpenShift image? OpenShift version is not enough?

Reply:
Because the OpenShift release image is the SSOT we use for other images like MCO, must-gather, etc.

Reply:
If I am not mistaken the OLM can't work on disconnected, because we don't create our own catalog source. So this won't work on disconnected as well as insstalling any OLM operator.